packages/cli/src/ui/components/shared/text-buffer.ts

```diff
@@ -1281,13 +1251,41 @@ export function useTextBuffer({
     [visualLines, visualScrollRow, viewport.height],
   );
 
+  const replaceRange = useCallback(
+    (
+      startRow: number,
+      startCol: number,
+      endRow: number,
+      endCol: number,
+      text: string,
+    ): boolean => {
+      if (
+        startRow > endRow ||
+        (startRow === endRow && startCol > endCol) ||
+        startRow < 0 ||
+        startCol < 0 ||
+        endRow >= lines.length ||
+        (endRow < lines.length && endCol > cpLen(lines[endRow]))
+      ) {
+        return false;
+      }
+      dispatch({
```

Comment:
remove the code to `return false`.
should massively simplify this to just forward to calling dispatch.